### PR TITLE
[REEF-1261] Enable ConstructorWithoutParamsCheck from sevntu-checkstyle

### PR DESCRIPTION
**Pull request by zerg-junior**
This patch:
 * Bumps up the sevntu-checkstyle version to 1.20.0; the check is released in this version
 * Fixes violations of the check in the REEF Java codebase
 * Makes the check ignore the test files

JIRA:
 [REEF-1261] (https://issues.apache.org/jira/browse/REEF-1261)

### CODE REVIEW

**Comment by zerg-junior** [comment]
Please proofread exception messages.  I haven't worked on many parts of code this check touches, so it's hard for me to come up with meaningful messages.

File: lang/java/reef-tang/tang-tint/src/main/java/org/apache/reef/tang/util/Tint.java [hunk 1]
```diff
@@ -362,7 +362,8 @@ public static void main(final String[] args)
           } else if (n instanceof ClassNode<?>) {
             out.println(t.toHtmlString((ClassNode<?>) n, currentPackage));
           } else {
-            throw new IllegalStateException();
+            throw new IllegalStateException("The node is neither CLassNode nor NamedParameterNode."
```
Comment (markusweimer) [inline lang/java/reef-tang/tang-tint/src/main/java/org/apache/reef/tang/util/Tint.java]: Typo: `CLassNode` --> `ClassNode`

File: lang/java/reef-tang/tang/src/main/java/org/apache/reef/tang/implementation/Subplan.java [hunk 2]
```diff
@@ -32,7 +32,8 @@ public Subplan(final Node node, final int selectedIndex, final InjectionPlan<T>.
     super(node);
     this.alternatives = alternatives;
     if (selectedIndex < -1 || selectedIndex >= alternatives.length) {
-      throw new ArrayIndexOutOfBoundsException();
+      throw new ArrayIndexOutOfBoundsException("Actual value of  selectedIndex is " + selectedIndex
```
Comment (markusweimer) [inline lang/java/reef-tang/tang/src/main/java/org/apache/reef/tang/implementation/Subplan.java]: One too many spaces in front of `selectedIndex`

File: lang/java/reef-tang/tang/src/main/java/org/apache/reef/tang/implementation/Subplan.java [hunk 3]
```diff
@@ -127,7 +128,7 @@ public int getSelectedIndex() {
 
   public InjectionPlan<? extends T> getDelegatedPlan() {
     if (selectedIndex == -1) {
-      throw new IllegalStateException();
+      throw new IllegalStateException("selectedIndex equals " + selectedIndex);
```
Comment (markusweimer) [inline lang/java/reef-tang/tang/src/main/java/org/apache/reef/tang/implementation/Subplan.java]: We know what it is: `-1`. No need for the string concat.

File: lang/java/reef-tests/src/test/java/org/apache/reef/tests/evaluatorfailure/FailureSchedulingContextStartHandler.java [hunk 4]
```diff
@@ -40,6 +40,7 @@
   }
 
   @Override
+  @SuppressWarnings("checkstyle:constructorwithoutparams")
```
Comment (markusweimer) [inline lang/java/reef-tests/src/test/java/org/apache/reef/tests/evaluatorfailure/FailureSchedulingContextStartHandler.java]: Why are we suppressing the warning here?

**Comment by markusweimer** [comment]
Thanks! I have done a pass and found some minor nitpicks.

**Comment by zerg-junior** [comment]
@markusweimer Updated.

**Comment by markusweimer** [comment]
LGTM. @dafrista Can you have a look at the vortex messages?

**Comment by johnyangk** [comment]
The vortex messages look good to me.

**Comment by markusweimer** [comment]
Thanks, @johnyangk, I will merge

**Comment by markusweimer** [comment]
I have failed tests with this, after I rebase onto `master`:

```
Failed tests:
  TestMultipleCommGroups.testMultipleCommGroups:72 Job state after execution: FORCE_CLOSED
```

Looks like a timeout?

**Comment by zerg-junior** [comment]
Rebased to uspream/master.

**Comment by markusweimer** [comment]
I'll test again.

**Pull request closed without merging**